### PR TITLE
Refactor dev environment

### PR DESCRIPTION
**Pull request by joewest**
This change adds several key things:
- Moves from golang to bash for the wizard 
- Removes the pre-created keys
- Introduces a pattern for how dev commands and shortcuts can be added to the Env

It does not consume the latest changes to `livepeer`, specifically the removal of `-ethDataDir` flag. that is a good thing to tackle next.

It also can get into a bit of a wonky state if you run multiple vms that use the same project src repos on your host machine.

### Why bash?

> "What was the inspiration to move the wizard from go to bash? Personal preference? All shell stuff anyway?" - @dob 

Two primary reasons:
1. I want people to be able to understand what the wizard is doing without parsing go code
2. All of the commands are effectively shell commands

Everything else kind of rolls up into those two. There's probably an overarching goal of deprecating the wizard as a primary tool. I think it is a bad middle ground currently -- it isn't flexible enough for devs and isn't easy enough for non-devs.

### Next Steps

I'm not entirely happy with some of the specific functions in `dot_lpdev_cmds.sh`, but I like the overall pattern and want to get this out for feedback. They're a little brittle right now and can probably be simplified.

The specific direction I was thinking is to make this tool more modular so it's easier to extend to as new dev needs arise. Eventually it could support various deployment schemes as well. Here is one possible direction:

```
$ lpdev geth [ init run reset ]
$ lpdev protocol [ init deploy reset ]
$ lpdev nodes [ --reset ]
$ lpdev node [ broadcaster transcoder ] <node-id> [ --reset ]
$ lpdev [ status wizard doctor reset ]
```

Once those commands exist and are robust, it should be pretty easy to replicate the wizard functionality with them.

### CODE REVIEW

**Comment by joewest** [comment]
Note: I'm not entirely happy with some of the specific functions in `dot_lpdev_cmds.sh`, but I like the overall pattern and want to get this out for feedback. They're a little brittle right now and can probably be simplified.

**Comment by dob** [comment]
I had to manually create `.lpdata` directory. Prior to that the commands in the wizard to start geth wouldn't complete.

**Comment by dob** [comment]
And geth is running, but when I try the deploy protocol contracts command I get the following errors:

```
== Current Status ==
  
Geth miner is running: true (2240)
Geth accounts:
  e2f3775121ba3d46e640374cdd41051a545f6367 (miner)

Protocol is built: true (current branch: master)

Broadcaster node is running: false ()
Transcoder node is running: false ()

--
  
What would you like to do?
1) Display status			4) Start & set up broadcaster node	7) Exit
2) Set up & start Geth local network	5) Start & set up transcoder node
3) Deploy/overwrite protocol contracts	6) Destroy current environment
#? 3
Protocol src directory exists
Geth Mining Account not found
Running `truffle migrate --reset --network lpTestNet`
Using network 'lpTestNet'.

Running migration: 1_initial_migration.js
  Deploying Contract...
  ... undefined
Error encountered, bailing. Network state unknown. Review successful transactions manually.
Error: unknown account
    at Object.InvalidResponse (/usr/lib/node_modules/truffle/build/cli.bundled.js:37312:16)
    at /usr/lib/node_modules/truffle/build/cli.bundled.js:220420:36
    at /usr/lib/node_modules/truffle/build/cli.bundled.js:72910:11
    at /usr/lib/node_modules/truffle/build/cli.bundled.js:204149:9
    at XMLHttpRequest.request.onreadystatechange (/usr/lib/node_modules/truffle/build/cli.bundled.js:205574:13)
    at XMLHttpRequestEventTarget.dispatchEvent (/usr/lib/node_modules/truffle/build/cli.bundled.js:73069:18)
    at XMLHttpRequest._setReadyState (/usr/lib/node_modules/truffle/build/cli.bundled.js:73359:12)
    at XMLHttpRequest._onHttpResponseEnd (/usr/lib/node_modules/truffle/build/cli.bundled.js:73514:12)
    at IncomingMessage.<anonymous> (/usr/lib/node_modules/truffle/build/cli.bundled.js:73474:24)
    at emitNone (events.js:110:20)
Don't forget to restart any nodes that may be using the previous controllerAddr!
```

**Comment by dob** [comment]
And then somehow on a rebuild of the environment, I could get beyond that point, but compilation of the contracts still failed because of an unrecognized `view` keyword. Perhaps it's using an older version of solidity.

```
/home/ubuntu/src/protocol/contracts/Controller.sol:28:46: ParserError: Expected token LBrace got 'View'
    function getContract(bytes32 _id) public view returns (address) {
                                             ^
,/home/ubuntu/src/protocol/contracts/Initializable.sol:23:37: ParserError: Expected token LBrace got 'View'
    function isInitialized() public view returns (bool) {

...
```

If these comments aren't helpful, let me know and we can take the troubleshooting offline. Just adding them here for review before we track them as independent issues in case I'm doing something wrong.

File: dot_lpdev_cmds.sh [hunk 1]
```diff
@@ -0,0 +1,719 @@
...
+    git clone -b develop "https://github.com/livepeer/protocol.git"
+    cd $OPWD
+  fi
+
+  ##
+  # Install local dev truffle.js
+  ##
+
+  __lpdev_geth_refresh_status
+  if [ -z "${gethMiningAccountngAccount}" ]
```
Comment (yondonfu) [inline dot_lpdev_cmds.sh]: "gethMiningAccount"?

File: dot_lpdev_cmds.sh [hunk 2]
```diff
@@ -0,0 +1,719 @@
...
+  else
+    echo "Running \`npm install\`"
+    OPWD=$PWD
+    cd $srcDir/protocol
+    npm install
+    cd $OPWD
+  fi
+}
+
+function __lpdev_protocol_deploy {
```
Comment (yondonfu) [inline dot_lpdev_cmds.sh]: Might need to include a check somewhere here to make sure the protocol init stuff like npm install didn't fail. I was running into an error when deploying due to the node modules not being installed

**Comment by joewest** [comment]
@dob please add comments here liberally so we can track and fix.

I'm generally running this as a clean environment -- blowing away each time and restarting from scratch so I can imagine some issues.

For instance, I just realized most devs moving from OSX to here will have built `protocol/node_modules` and any native modules won't run in the vm (boo).

File: dot_lpdev_cmds.sh (hunk 1 above)
Comment (joewest) [reply dot_lpdev_cmds.sh]: Embarrassing

File: dot_lpdev_cmds.sh (hunk 2 above)
Comment (joewest) [reply dot_lpdev_cmds.sh]: As @dob has discovered in his feedback, the npm part is going to be the weirdest. I'm going to see if there's a quick sane way to create a uniq node environment that doesn't depend on `protocol/node_modules`

**Comment by joewest** [comment]
Ok. Whew.

I took a stab at creating a faux `~/src/protocol/node_modules` in the virtual machine. Please give this a whirl. You should be able to `vagrant provision` or `vagrant up --provision` to get the latest changes without destroying your existing vm.

**Friendly FYI:** The `node_modules` directory in your host machine will no longer be two-way synced between the host and vm. Everything else should be.

It could be interesting to extend this idea to include anything else that is dev/vm specific (like truffle.js, etc).

**Comment by ericxtang** [comment]
Awesome.  Let's merge this PR and keep an improvement ideas in the issues!  🚢

**Pull request merged**